lib/Sema/CSSimplify.cpp

```diff
@@ -9841,7 +9825,20 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
 
   case FixKind::AllowArgumentTypeMismatch: {
     increaseScore(SK_Fix);
-    return recordFix(fix) ? SolutionKind::Error : SolutionKind::Solved;
+    
+    auto impact = 1;
```

Comment:
By default we'd assume that there is only one mismatch so impact should be `2`, if there is more impact should be adjusted to `+= 3;`.

Reply:
Ahh, so I think I misunderstood the comment... right let me adjust that then :)

Reply:
Adjusted!